### ISSUE TITLE
unused variable in ecdsa.c

### ISSUE THREAD

**Issue by trandinhkhoa** [Mbed-TLS/mbedtls#2629]
When building mbedtls with option **MBEDTLS_ECDSA_VERIFY_ALT** enabled (`_config.h_`

), compilation fails due to unused paramter  **rs_ctx** and the  `-Werror -Wall` flag passed to GCC.
```
gcc-linaro-7.1.1-2017.08-x86_64_arm-eabi/bin/arm-eabi-gcc -std=c99 -mthumb-interwork -fstack-usage -O3 -fdata-sections -ffunction-sections -fno-short-enums -nostdlib -DTA_STACK_SIZE=16384 -DTRUSTEDAPP  -Werror -Wall -Wextra -DPLAT=ARMV7_A_STD -DARM_ARCH=ARMv7 -D__ARMv7__ -D__ARMV7_A__ -D__ARMV7_A_STD__  -DTBASE_API_LEVEL=11 -U__INT32_TYPE__ -U__UINT32_TYPE__  -D__INT32_TYPE__="int" -fstack-protector-strong -fpie -DDEBUG --debug   -march=armv7-a -mfpu=vfp -mfloat-abi=soft -mthumb  

../Externals/mbedTLS/Out/Code/library/ecdsa.c: In function ‘mbedtls_ecdsa_read_signature_restartable’:
../Externals/mbedTLS/Out/Code/library/ecdsa.c:739:54: error: unused parameter ‘rs_ctx’ [-Werror=unused-parameter]
                           mbedtls_ecdsa_restart_ctx *rs_ctx )
                                                      ^~~~~~
cc1: all warnings being treated as errors
```
**mbed TLS build:**  
Version: 2.16.1

In function `mbedtls_ecdsa_read_signature_restartable()` (file `_ecdsa.c_`), `#if defined(MBEDTLS_ECDSA_VERIFY_ALT)`, then `rs_ctx` is not used at all

**Comment by trandinhkhoa** [comment]
When building mbedtls with option **MBEDTLS_ECDSA_VERIFY_ALT** enabled (`_config.h_`

), compilation fails due to unused paramter  **rs_ctx** and the  `-Werror -Wall` flag passed to GCC.
```
gcc-linaro-7.1.1-2017.08-x86_64_arm-eabi/bin/arm-eabi-gcc -std=c99 -mthumb-interwork -fstack-usage -O3 -fdata-sections -ffunction-sections -fno-short-enums -nostdlib -DTA_STACK_SIZE=16384 -DTRUSTEDAPP  -Werror -Wall -Wextra -DPLAT=ARMV7_A_STD -DARM_ARCH=ARMv7 -D__ARMv7__ -D__ARMV7_A__ -D__ARMV7_A_STD__  -DTBASE_API_LEVEL=11 -U__INT32_TYPE__ -U__UINT32_TYPE__  -D__INT32_TYPE__="int" -fstack-protector-strong -fpie -DDEBUG --debug   -march=armv7-a -mfpu=vfp -mfloat-abi=soft -mthumb  

../Externals/mbedTLS/Out/Code/library/ecdsa.c: In function ‘mbedtls_ecdsa_read_signature_restartable’:
../Externals/mbedTLS/Out/Code/library/ecdsa.c:739:54: error: unused parameter ‘rs_ctx’ [-Werror=unused-parameter]
                           mbedtls_ecdsa_restart_ctx *rs_ctx )
                                                      ^~~~~~
cc1: all warnings being treated as errors
```
**mbed TLS build:**  
Version: 2.16.1

In function `mbedtls_ecdsa_read_signature_restartable()` (file `_ecdsa.c_`), `#if defined(MBEDTLS_ECDSA_VERIFY_ALT)`, then `rs_ctx` is not used at all

**Comment by RonEld** [comment]
@trandinhkhoa Thank you for reporting this issue!
Will we look into this.

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-2845

**Comment by tom-cosgrove-arm** [comment]
This was fixed by #4008
